cmake/CompilerConfiguration.cmake

```diff
@@ -41,7 +41,7 @@ Please open an issue at https://github.com/cp2k/dbcsr/issues with the reported c
 endif ()
 
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-  set(CMAKE_CXX_FLAGS_RELEASE      "-O3 -g -funroll-loops -Wall -Werror")
```

Comment:
Forgot to reply, but this was suggested above by @dev-zero in the review. I can bring it back, no problem. Maybe it's not very convenient though if compilers we don't test generate warnings and fail to compile as a result.

Reply:
This is actually a good point.. @dev-zero is there a way to add this flag for the CSCS tests? Something during the cmake call...

Reply:
If timing permits, we should run the CI in debug mode, rather than keep `-Werror` in release mode. The rational behind this comes from my distro work: compilers usually get stricter over time with regard to their warnings, hence things tend to break. For the user a warning is a signal that he might want to check for a newer version (where we ideally already fixed that issue since we test in debug), while it still allows him to build this specific version since he may depend on its exact behaviour.
Hence many source-based distros (Gentoo, Arch AUR) tend to strip the `-Werror`.

To specify custom flags we [currently recommend](https://cp2k.github.io/dbcsr/develop/page/2-user-guide/1-installation/1-cmake-build-recipes.html) to use a custom CMake build type:

```
cmake \
         -DCMAKE_BUILD_TYPE=custom \
         -DCMAKE_C_FLAGS_CUSTOM="-O3 -march=native" \
         -DCMAKE_CXX_FLAGS_CUSTOM="-O3 -march=native" \
         -DCMAKE_Fortran_FLAGS_CUSTOM="-O3 -march=native" \
         ..
```
... this is again something I've seen Linux distros doing: to inject their optimization flags, paths, etc. they started to define custom build types for CMake.
But this would mean we'd have to replicate the currently present DEBUG flags.

Reply:
Then no, let's lave -Werror in the release...

Reply:
Another option is to add a new build type `ReleaseStrict` that inherits all flags from `Release` and adds `-Werror`, but to me it seems like a lot of boilerplate for something trivial. I think easiest is when managers add a patch to drop `-Werror`.